apps/web/src/pages/templates/components/SmsPreview.tsx

```diff
@@ -0,0 +1,95 @@
+import styled from '@emotion/styled';
+import { colors } from '@novu/design-system';
+import { useAuthController, useDataRef } from '@novu/shared-web';
+import { useEffect, useState } from 'react';
+import { useFormContext, useWatch } from 'react-hook-form';
+
+import { useGetLocalesFromContent, usePreviewSms } from '../../../api/hooks';
+import { LocaleSelect } from '../../../components/workflow/Preview/common';
+import { useNavigateToStepEditor } from '../hooks/useNavigateToStepEditor';
+import { useStepFormErrors } from '../hooks/useStepFormErrors';
+import { useStepFormPath } from '../hooks/useStepFormPath';
+import { IForm } from './formTypes';
+import { MobileSimulator } from './phone-simulator';
+import { SmsBubble } from './SmsBubble';
+
+const BodyContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+  margin: auto 20px 40px 20px;
+`;
+
+const LocaleSelectStyled = styled(LocaleSelect)`
+  .mantine-Select-input {
+    color: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.white : colors.B60)};
+  }
+
+  .mantine-Input-rightSection {
+    color: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.white : colors.B60)} !important;
+  }
+`;
+
+export const SmsPreview = () => {
+  const { control } = useFormContext<IForm>();
+  const path = useStepFormPath();
+  const error = useStepFormErrors();
+  const templateContent = useWatch({
+    name: `${path}.template.content`,
+    control,
+  });
+  const { organization } = useAuthController();
+  const [previewContent, setPreviewContent] = useState(templateContent as string);
+  const [selectedLocale, setSelectedLocale] = useState('');
+  const previewData = useDataRef({ templateContent });
+  const templateContentError = error?.template?.content?.message;
+
+  const { data: locales, isLoading: areLocalesLoading, getLocalesFromContent } = useGetLocalesFromContent();
+  const { isLoading: isPreviewContentLoading, getSmsPreview } = usePreviewSms({
+    onSuccess: (result) => {
+      setPreviewContent(result.content);
+    },
+  });
+
+  const { navigateToStepEditor } = useNavigateToStepEditor();
+
+  useEffect(() => {
+    getLocalesFromContent({
+      content: previewData.current.templateContent,
+    });
+  }, [getLocalesFromContent, previewData]);
```

Comment:
fetch locales from the template, will be called only once during the initial mount